### PR TITLE
Added `databricks labs ucx create-table-mapping` and `databricks labs ucx manual-workspace-info` commands for CLI

### LINKED ISSUE #668: Add `databricks labs ucx manual-workspace-info` command
`databricks labs ucx manual-workspace-info --workspace-profiles profile-one,profile-two,profile-three --file x.json` to manually upload workspace info to a given set of workspaces, if teams didn't manage to get admins to run [the command](https://github.com/databrickslabs/ucx/issues/667)

### LINKED ISSUE #669: Add `databricks labs ucx table-mapping` command
databricks labs ucx table-mapping to create table mapping CSV file with the following columns: src_workspace_name,dst_catalog,src_schema,dst_schema,src_table,dst_table and let it be reviewed by people offline

Related issues:
- https://github.com/databrickslabs/ucx/issues/323
- https://github.com/databrickslabs/ucx/issues/304

Upstream dependencies that produce artifacts:
- https://github.com/databrickslabs/ucx/issues/667
- https://github.com/databrickslabs/ucx/issues/668

### PR DESCRIPTION
**Pull request by nfx**
Closes #668
Closes #669

### CODE REVIEW

File: src/databricks/labs/ucx/hive_metastore/mapping.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+class WorkspaceInfoReader:
+    def __init__(self, ws: WorkspaceClient, folder: str):
+        self._ws = ws
+        self._folder = folder
+
+    def _current_workspace_id(self) -> int:
+        headers = self._ws.config.authenticate()
+        headers["User-Agent"] = self._ws.config.user_agent
+        response = requests.get(f"{self._ws.config.host}/api/2.0/preview/scim/v2/Me", headers=headers, timeout=10)
```
Comment (dmoore247) [inline src/databricks/labs/ucx/hive_metastore/mapping.py:25]: I think there is a SDK for getting `me`, don't use requests.

File: src/databricks/labs/ucx/hive_metastore/mapping.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    def _load_workspace_info(self) -> dict[int, Workspace]:
+        try:
+            id_to_workspace = {}
+            workspace_info = self._ws.workspace.download(f"{self._folder}/workspace-info.json")
+            for workspace_metadata in json.loads(workspace_info):
+                workspace = Workspace.from_dict(workspace_metadata)
+                id_to_workspace[workspace.workspace_id] = workspace
+            return id_to_workspace
+        except NotFound:
```
Comment (dmoore247) [inline src/databricks/labs/ucx/hive_metastore/mapping.py:36]: Log the exception, as the root cause could have something other than `NotFound`
What if you have permission denied?

Comment (dmoore247) [reply src/databricks/labs/ucx/hive_metastore/mapping.py:36]: Need unit and integration tests

File: src/databricks/labs/ucx/hive_metastore/mapping.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+class WorkspaceInfoReader:
+    def __init__(self, ws: WorkspaceClient, folder: str):
+        self._ws = ws
+        self._folder = folder
+
+    def _current_workspace_id(self) -> int:
+        headers = self._ws.config.authenticate()
+        headers["User-Agent"] = self._ws.config.user_agent
+        response = requests.get(f"{self._ws.config.host}/api/2.0/preview/scim/v2/Me", headers=headers, timeout=10)
+        return int(response.headers.get("x-databricks-org-id"))
```
Comment (dmoore247) [inline src/databricks/labs/ucx/hive_metastore/mapping.py:26]: This should be found in the ws config object somewhere.
`int()` casting operations need a lot of exception handling.
Add unit and integration testing.

File: src/databricks/labs/ucx/hive_metastore/mapping.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            msg = "Please run as account-admin: databricks labs ucx sync-workspace-info"
+            raise ValueError(msg) from None
+
+    def current(self) -> str:
+        workspace_id = self._current_workspace_id()
+        workspaces = self._load_workspace_info()
+        if workspace_id not in workspaces:
+            msg = f"Current workspace is not known: {workspace_id}"
+            raise KeyError(msg) from None
+        return workspaces[workspace_id].workspace_name
```
Comment (dmoore247) [inline src/databricks/labs/ucx/hive_metastore/mapping.py:46]: Definitely need unit test coverage

**Review by dmoore247 — CHANGES_REQUESTED** [review]
Please don't use requests, see project charter.
Please add more tests, and exception handling around `int()` (better don't implement requests)

File: src/databricks/labs/ucx/hive_metastore/mapping.py (hunk 3 above)
Comment (nfx) [reply src/databricks/labs/ucx/hive_metastore/mapping.py:26]: It's not there yet

**Pull request merged**